src/resolve/resolved-varlink.c

```diff
@@ -1228,6 +1251,43 @@ static int vl_method_resolve_record(Varlink *link, JsonVariant *parameters, Varl
         return 1;
 }
 
+static int vl_method_start_browse(Varlink* link, JsonVariant* parameters, VarlinkMethodFlags flags, void* userdata) {
+        static const JsonDispatch dispatch_table[] = {
+                { "domainName", JSON_VARIANT_STRING,   json_dispatch_string, offsetof(LookupParametersMdnsBrowse, domainName), JSON_MANDATORY },
+                { "name",       JSON_VARIANT_STRING,   json_dispatch_string, offsetof(LookupParametersMdnsBrowse, name),       JSON_MANDATORY },
+                { "type",       JSON_VARIANT_STRING,   json_dispatch_string, offsetof(LookupParametersMdnsBrowse, type),       JSON_MANDATORY },
+                { "ifindex",    JSON_VARIANT_INTEGER,  json_dispatch_int,    offsetof(LookupParametersMdnsBrowse, ifindex),    JSON_MANDATORY },
+                { "flags",      JSON_VARIANT_UNSIGNED, json_dispatch_uint64, offsetof(LookupParametersMdnsBrowse, flags),      JSON_MANDATORY },
+                {}
+        };
+
+        _cleanup_(lookup_parameters_mdns_destroy) LookupParametersMdnsBrowse p = {};
+        Manager *m;
+        int r = 0;
+
+        assert(link);
+
+        /* if the client didn't set the more flag, it is using us incorrectly */
+        if (!FLAGS_SET(flags, VARLINK_METHOD_MORE))
+                return varlink_error_invalid_parameter(link, NULL);
```

Comment:
In developer mode resolved crashes here on receiving method calls without the "more" flag:
```sh
printf '{"method":"io.systemd.Resolve.StartBrowse", "parameters":{"domainName":"","name":"","type":"","ifindex":0,"flags":0}}\0' | ncat -U /run/systemd/resolve/io.systemd.Resolve
```
```sh
(gdb) bt
#0  0x00007f0430eae834 in __pthread_kill_implementation () from /lib64/libc.so.6
#1  0x00007f0430e5c8ee in raise () from /lib64/libc.so.6
#2  0x00007f0430e448ff in abort () from /lib64/libc.so.6
#3  0x00007f0432924e31 in log_assert_failed (text=0x7f043308c320 <str> "parameters", file=0x7f04330896e3 "src/shared/varlink.c", line=2622, 
    func=0x7f043308d040 <__func__.varlink_error_invalid_parameter> "varlink_error_invalid_parameter") at ../src/basic/log.c:968
#4  0x00007f043292bec7 in log_assert_failed_return (text=0x7f043308c320 <str> "parameters", file=0x7f04330896e3 "src/shared/varlink.c", line=2622, 
    func=0x7f043308d040 <__func__.varlink_error_invalid_parameter> "varlink_error_invalid_parameter") at ../src/basic/log.c:987
#5  0x00007f043271ac71 in varlink_error_invalid_parameter (v=0x51200002ac40, parameters=0x0) at ../src/shared/varlink.c:2622
#6  0x000000000065b3ca in vl_method_start_browse (link=0x51200002ac40, parameters=0x511000055720, flags=0, userdata=0x0)
    at ../src/resolve/resolved-varlink.c:1272
#7  0x00007f04326f7177 in varlink_dispatch_method (v=0x51200002ac40) at ../src/shared/varlink.c:1464
#8  0x00007f04326ef316 in varlink_process (v=0x51200002ac40) at ../src/shared/varlink.c:1537
#9  0x00007f0432723201 in defer_callback (s=0x50e0000217c0, userdata=0x51200002ac40) at ../src/shared/varlink.c:2926
#10 0x00007f0432f35318 in source_dispatch (s=0x50e0000217c0) at ../src/libsystemd/sd-event/sd-event.c:4219
#11 0x00007f0432f322d5 in sd_event_dispatch (e=0x517000022280) at ../src/libsystemd/sd-event/sd-event.c:4808
#12 0x00007f0432f37dce in sd_event_run (e=0x517000022280, timeout=18446744073709551615) at ../src/libsystemd/sd-event/sd-event.c:4869
#13 0x00007f0432f38d5a in sd_event_loop (e=0x517000022280) at ../src/libsystemd/sd-event/sd-event.c:4891
#14 0x0000000000683089 in run (argc=1, argv=0x7ffe45fc1fe8) at ../src/resolve/resolved.c:92
#15 0x00000000006810de in main (argc=1, argv=0x7ffe45fc1fe8) at ../src/resolve/resolved.c:99
```

(Looking at 43bd70d58450e03eabbcdf4265f4099353b6b40f I'm guessing it was borrowed from `io.systemd.Resolve.Monitor.SubscribeQueryResults`. It crashes in developer mode too)

Reply:
Looks like it was fixed.